### ISSUE TITLE
Type-Incompatibility of `round()` when used in compound-statements

### ISSUE THREAD

**Issue by chadhpaine** [python/mypy#16991]
**Bug Report**

When calls to `round()` are used in some compound statements - especially those that resolve to booleans - `mypy` reports an `[arg-type]` violation.

**To Reproduce**

```python
x: float = 1.333
reveal_type(round(x))  # int  # ok
reveal_type(3 and round(x))  # int  # ok
reveal_type(True and round(x))  # bool  # error: Argument 1 to "round" has incompatible type "float"; expected "_SupportsRound1[bool]"  [arg-type]
```

**Expected Behavior**

`mypy` should not report `[arg-type]`.

**Actual Behavior**

```
main.py:2: note: Revealed type is "builtins.int"
main.py:3: note: Revealed type is "builtins.int"
main.py:4: note: Revealed type is "builtins.bool"
main.py:4: error: Argument 1 to "round" has incompatible type "float"; expected "_SupportsRound1[bool]"  [arg-type]
main.py:4: note: Following member(s) of "float" have conflicts:
main.py:4: note:     Expected:
main.py:4: note:         def __round__(self) -> bool
main.py:4: note:     Got:
main.py:4: note:         @overload
main.py:4: note:         def __round__(self, None = ..., /) -> int
main.py:4: note:         @overload
main.py:4: note:         def __round__(self, SupportsIndex, /) -> float
Found 1 error in 1 file (checked 1 source file)
```

`mypy` reports `[arg-type]`.

**Your Environment**

- Mypy version used: master branch via mypy-playground
- Mypy command-line flags: n/a
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: 3.12

**Comment by chadhpaine** [comment]
Appears to be resolved in mypy 3.20 / python 3.12 on mypy-play

```
x: float = 1.333
reveal_type(round(x))  # int  # ok
reveal_type(3 and round(x))  # int  # ok
reveal_type(True and round(x))  # int  # ok
reveal_type(round(x) and True)  # Literal[0] | Literal[True]?  # ok
```
